src/components/Invoice/EditInvoice.tsx

```diff
@@ -0,0 +1,19 @@
+import { useParams } from 'react-router-dom';
+import InvoiceForm from './InvoiceForm';
+import { useInvoice } from './use-invoice';
+import type { IInvoiceForm } from '../Models/InvoiceForm.interface';
+
+export default function EditInvoice() {
+  const { id } = useParams();
+  const { data } = useInvoice(id as string);
+  const invoice = data as IInvoiceForm;
+
+  if (!invoice) {
+    return <div>Loading</div>;
+  }
+
+  invoice.createdAt = new Date(invoice.createdAt as unknown as string);
```

Comment:
If you add correct zod validation, you wouldn't need these 2 lines

Reply:
I need them to convert data, without this I would get
![image](https://github.com/broslukasz/react-upskill-basics/assets/14224110/48c86c36-3f6d-4986-a2da-2c1dd9695e95)

Reply:
removed